From http://xenbits.xen.org/hg/linux-2.6.18-xen.hg/rev/87faef7f0177
From: xen-devel@lists.xenproject.org
Patch-mainline: Never, SUSE-Xen specific
Subject: netfront: don't truncate grant references
References: none

They're of type uint32_t, and hence error checks must not use a more
narrow type.

Reported-by: Dongli Zhang <dongli.zhang@oracle.com>
Signed-off-by: Jan Beulich <jbeulich@suse.com>

--- a/drivers/xen/netfront/netfront.c
+++ b/drivers/xen/netfront/netfront.c
@@ -798,7 +798,7 @@ no_skb:
 		np->rx_skbs[id] = skb;
 
 		ref = gnttab_claim_grant_reference(&np->gref_rx_head);
-		BUG_ON((signed short)ref < 0);
+		BUG_ON(IS_ERR_VALUE((int32_t)ref));
 		np->grant_rx_ref[id] = ref;
 
 		pfn = page_to_pfn(skb_shinfo(skb)->frags[0].page);
@@ -905,7 +905,7 @@ static void xennet_make_frags(struct sk_
 		tx = RING_GET_REQUEST(&np->tx, prod++);
 		tx->id = id;
 		ref = gnttab_claim_grant_reference(&np->gref_tx_head);
-		BUG_ON((signed short)ref < 0);
+		BUG_ON(IS_ERR_VALUE((int32_t)ref));
 
 		mfn = virt_to_mfn(data);
 		gnttab_grant_foreign_access_ref(ref, np->xbdev->otherend_id,
@@ -927,7 +927,7 @@ static void xennet_make_frags(struct sk_
 		tx = RING_GET_REQUEST(&np->tx, prod++);
 		tx->id = id;
 		ref = gnttab_claim_grant_reference(&np->gref_tx_head);
-		BUG_ON((signed short)ref < 0);
+		BUG_ON(IS_ERR_VALUE((int32_t)ref));
 
 		mfn = pfn_to_mfn(page_to_pfn(frag->page));
 		gnttab_grant_foreign_access_ref(ref, np->xbdev->otherend_id,
@@ -1000,7 +1000,7 @@ static int network_start_xmit(struct sk_
 
 	tx->id   = id;
 	ref = gnttab_claim_grant_reference(&np->gref_tx_head);
-	BUG_ON((signed short)ref < 0);
+	BUG_ON(IS_ERR_VALUE((int32_t)ref));
 	mfn = virt_to_mfn(data);
 	gnttab_grant_foreign_access_ref(
 		ref, np->xbdev->otherend_id, mfn, GTF_readonly);
